### PR TITLE
Copter:Skip unnecessary pre-arm compass checks

### PR DESCRIPTION
**Pull request by arshPratap**
Skip the unnecessary compass pre-arm checks as mentioned here #16513.Thanks

### CODE REVIEW

**Review by amilcarlucas — APPROVED** [review]

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -631,7 +631,8 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
 
 #ifndef ALLOW_ARM_NO_COMPASS
     // if external source of heading is available, we can skip compass health check
-    if (!ahrs.is_ext_nav_used_for_yaw()) {
+    // skip this check if EKF's yaw source is not compass
+    if (!ahrs.is_ext_nav_used_for_yaw() && ahrs.EKF3.isYaw_Source(AP_NavEKF_Source::SourceYaw::COMPASS)) {
```
Comment (tridge) [inline ArduCopter/AP_Arming.cpp:635]: needs to cope with EKF3 not in the build (use HAL_NAVEKF3_AVAILABLE) and also needs to cope with SourceYaw::GPS_COMPASS_FALLBACK and also needs to check we are actually using EKF3

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduCopter/AP_Arming.cpp (hunk 1 above)
Comment (tridge) [reply ArduCopter/AP_Arming.cpp:635]: better to add a AP_AHRS::need_compass() call

Comment (arshPratap) [reply ArduCopter/AP_Arming.cpp:635]: Got it ,sir.I will start working on it immediately.

**Comment by peterbarker** [comment]
@arshPratap this now needs a rebase.  Are you still interested in pursuing this?

**Comment by peterbarker** [comment]
No answer in 3 years - closing assuming @arshPratap isn't interested in this one.  Feel free to create a new PR if you think there's still an issue!

**Pull request closed without merging**